src/renderer/navigation/history.ts

```diff
@@ -1,6 +1,15 @@
 import { ipcRenderer } from "electron";
 import { createBrowserHistory, createMemoryHistory } from "history";
 import { createObservableHistory } from "mobx-observable-history";
+import { getHostedClusterId } from "../../common/cluster-store";
+import logger from "../../main/logger";
 
 export const history = ipcRenderer ? createBrowserHistory() : createMemoryHistory();
 export const navigation = createObservableHistory(history);
+
+navigation.listen((location, action) => {
+  const clusterId = getHostedClusterId();
+  const frame = clusterId ?? "root";
+
+  logger.debug(`[NAVIGATION]: ${action}`, { location, frame });
+});
```

Comment:
Do we really need to know `clusterId` in debug info?
Otherwise I would rather to current `location.href` which is more informative (and contains cluster-id) + no need in import `cluster-store.ts`

```suggestion
navigation.listen((location, action) => {
  const isClusterView = !process.isMainFrame;

  logger.debug(`[NAVIGATION]: ${action}`, { domain: location.href, location, isClusterView });
});
```

Reply:
I found parsing that object harder to read but sure we can have this instead.